File: TeamCodeC/src/main/java/org/firstinspires/ftc/teamcodec/subsystem/Intake.java
package org.firstinspires.ftc.teamcodec.subsystem;

import com.qualcomm.robotcore.hardware.HardwareMap;
import com.seattlesolvers.solverslib.command.SubsystemBase;
import com.seattlesolvers.solverslib.hardware.motors.Motor;

public class Intake extends SubsystemBase {
    private final Motor intakeMotor;
    private boolean isRunning = false;

    public Intake(final HardwareMap hMap) {
        intakeMotor = new Motor(hMap, "intake", Motor.GoBILDA.BARE);
        intakeMotor.setRunMode(Motor.RunMode.RawPower);
        System.out.println("IntakeSubsystem initialized");
    }

    public void toggle() {
        isRunning = !isRunning;
    }

    public void setPower(double power) {
        intakeMotor.set(power);
    }
    public void start() {
        System.out.println("Intake started");
        intakeMotor.set(1.0);
    }

    public void stop() {
        intakeMotor.set(0.0);
    }

    public boolean active() {
        return isRunning;
    }
}

File: TeamCodeD/src/main/java/org/firstinspires/ftc/teamcode/hardware/Intake.kt
package org.firstinspires.ftc.teamcode.hardware

import com.bylazar.telemetry.TelemetryManager
import com.qualcomm.robotcore.hardware.DcMotor
import com.qualcomm.robotcore.hardware.HardwareMap
import org.firstinspires.ftc.teamcode.constants.IntakeConstants


class Intake(hardwareMap: HardwareMap) {
    private val intakeMotor = hardwareMap.dcMotor["intake"]

    init {
        intakeMotor.mode = DcMotor.RunMode.RUN_WITHOUT_ENCODER
        intakeMotor.zeroPowerBehavior = DcMotor.ZeroPowerBehavior.BRAKE
        intakeMotor.power = 0.0
    }

    var active = false

    fun start() {
        active = true
    }

    fun stop() {
        active = false
    }

    fun teleOpPeriodic(driverStrength: Double, telemetry: TelemetryManager) {
        intakeMotor.power = if (active) IntakeConstants.ON_POWER else driverStrength

        telemetry.addData("Intake active", active)
        telemetry.addData("Intake power", intakeMotor.power)
    }

    fun periodic( telemetry: TelemetryManager) {
        intakeMotor.power = if (active) IntakeConstants.ON_POWER else IntakeConstants.OFF_POWER
        telemetry.addData("Intake active", active)
        telemetry.addData("Intake power", intakeMotor.power)
    }
}